File: jung-algorithms/src/main/java/edu/uci/ics/jung/layout/util/LayoutEventSupport.java
package edu.uci.ics.jung.layout.util;

/**
 * interface for support to LayoutChangeListeners
 *
 * @param <N>
 */
public interface LayoutEventSupport<N> {

  void addLayoutChangeListener(LayoutChangeListener<N> listener);

  void removeLayoutChangeListener(LayoutChangeListener<N> listener);
}

File: jung-algorithms/src/main/kotlin/edu/uci/ics/jung/layout/util/LayoutEventSupport.kt
package edu.uci.ics.jung.layout.util

/**
 * interface for support to LayoutChangeListeners
 *
 * @param N
 */
interface LayoutEventSupport<N : Any> {

  fun addLayoutChangeListener(listener: LayoutChangeListener<N>)

  fun removeLayoutChangeListener(listener: LayoutChangeListener<N>)
}
